Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp

```diff
@@ -0,0 +1,264 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <libudev.h>
+#include <unistd.h>
+
+#include "Common/Logging/Log.h"
+#include "InputCommon/ControllerInterface/evdev/evdev.h"
+
+
+namespace ciface
+{
+namespace evdev
+{
+
+void Init(std::vector<Core::Device*> &controllerDevices)
+{
+	int num_controllers = 0;
+
+	// We use Udev to find any devices. In the future this will allow for hotplugging.
+	// But for now it is essentially iterating over /dev/input/event0 to event31. However if the
+	// naming scheme is ever updated in the future, this *should* be fowards compatable.
+
+	struct udev *udev = udev_new();
```

Comment:
`struct` is necessary, to distinguish between the `udev` pointer and the `struct udev` type.

No, I will not rename it.